tests/qat/test_core.py

```diff
@@ -113,6 +122,25 @@ def equivalent_vars(self, other):
     return isinstance(self, type(other)) and (vars(self) == vars(other))
 
 
+def equivalent_array(self, other):
+    if isinstance(self, type(other)):
+        vars_self = vars(self)
+        vars_other = vars(other)
+        if len(vars_self) != len(vars_other):
+            return False
+        for key in vars_self.keys():
+            if not key in vars_other:
+                return False
+            check = vars_self[key] == vars_other[key]
+            if not isinstance(check, bool):
+                check = all(check)
+                print(vars_self[key])
+                print(vars_other[key])
+            if not check:
+                return False
+    return True
```

Comment:
This seems to have the wrong indentation. `equivalent_array("Hello", 42)` looks to evaluate to `True`. :thinking: